services/director-v2/tests/unit/with_swarm/test_modules_dynamic_sidecar_docker_api.py

```diff
@@ -231,6 +232,36 @@ async def cleanup_dynamic_sidecar_stack(
         )
 
 
+@pytest.fixture
+async def project_id_labeled_network(
+    async_docker_client: aiodocker.docker.Docker, project_id: ProjectID
+) -> AsyncIterable[str]:
+    network_config = {
+        "Name": "test_network_by_project_id",
+        "Driver": "overlay",
+        "Labels": {"project_id": f"{project_id}"},
+    }
+    network_id = await docker_api.create_network(network_config)
+
+    yield network_id
+
+    network = await async_docker_client.networks.get(network_id)
+    await network.delete()
```

Comment:
you might need to use tenacity to ensure the network is really gone right?

Reply:
No, but an assert was missing. This is sufficient. See also reply below